### PR TITLE
Fix flake where cross ns finalizer can't be removed

### PR DESCRIPTION
**Pull request by akalenyu**
Signed-off-by: Alex Kalenyuk <akalenyu@redhat.com>

**What this PR does / why we need it**:
We fail to remove the finalizer because the mutating webhook validates source ns exists,
but source ns might not be around anymore, and that is absolutely fine if DV is marked for deletion.
The "offending" test case is `[test_id:4960]`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2363/pull-containerized-data-importer-e2e-ceph-gc/1552213098473984000
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2381/pull-containerized-data-importer-e2e-ceph-gc/1555211218673733632
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2371/pull-containerized-data-importer-e2e-ceph-gc/1553016969739898880

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apiserver/webhooks/datavolume-mutate.go [hunk 1]
```diff
@@ -130,9 +130,12 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
...
 
-	_, err := wh.k8sClient.CoreV1().Namespaces().Get(context.TODO(), sourceNamespace, metav1.GetOptions{})
-	if err != nil {
-		return toAdmissionResponseError(err)
+	if dataVolume.GetDeletionTimestamp() == nil {
+		// No point validating source ns exists if DV is flagged for deletion
+		_, err := wh.k8sClient.CoreV1().Namespaces().Get(context.TODO(), sourceNamespace, metav1.GetOptions{})
+		if err != nil {
+			return toAdmissionResponseError(err)
+		}
```
Comment (mhenriks) [inline pkg/apiserver/webhooks/datavolume-mutate.go:138]: Maybe just return success if deletion timestamp set.  No need to bother with the rest of this function, which I think is just about setting the clone token

Comment (akalenyu) [reply pkg/apiserver/webhooks/datavolume-mutate.go]: Done

**Comment by awels** [comment]
How about a unit test making sure this is working as expected, like `Should always accept mutate when dv is marked for deletion`?

**Comment by akalenyu** [comment]
> How about a unit test making sure this is working as expected, like `Should always accept mutate when dv is marked for deletion`?

good idea, done

**Comment by awels** [comment]
/lgtm
@mhenriks any other thoughts?

**Pull request merged**